lib/basedriver/commands/find.js

```diff
@@ -47,6 +65,120 @@ commands.findElementsFromElement = async function (strategy, selector, elementId
   return await this.findElOrElsWithProcessing(strategy, selector, true, elementId);
 };
 
+helpers.findByImage = async function (b64Template, {checkingStaleness = false} = {}) {
+  let threshold = DEFAULT_MATCH_THRESHOLD;
+  if (this.settings) {
+    threshold = this.settings.getSettings().imageMatchThreshold;
+  }
+  log.info(`Finding image element with match threshold ${threshold}`);
+  const {width: screenWidth, height: screenHeight} = await this.getWindowSize();
+
+  // someone might have sent in a template that's larger than the screen
+  // dimensions. If so let's check and cut it down to size since the algorithm
+  // will not work unless we do
+  b64Template = await this.ensureTemplateSize(b64Template, screenWidth, screenHeight);
+
+  let rect = null;
+  try {
+    await this.implicitWaitForCondition(async () => {
+      let b64Screenshot = await this.getScreenshotForImageFind(screenWidth, screenHeight);
+      rect = (await this.compareImages(MATCH_TEMPLATE_MODE, b64Screenshot,
+        b64Template, {threshold})).rect;
+      return true;
+    });
+  } catch (err) {
+    if (!err.match(/Condition unmet/)) {
+      throw err;
+    }
+  }
+
+  if (!rect) {
+    throw new errors.NoSuchElementError(); // eslint-disable-line no-unsafe-finally
+  }
+
+  log.info(`Image template matched: ${JSON.stringify(rect)}`);
+  const imgEl = new ImageElement(b64Template, rect);
+
+  // if we're just checking staleness, return straightaway so we don't add
+  // a new element to the cache
+  if (checkingStaleness) {
+    return imgEl;
+  }
+
+  this._imgElCache.add(imgEl);
+  const protoKey = this.isW3CProtocol() ? W3C_ELEMENT_KEY : MJSONWP_ELEMENT_KEY;
+  return imgEl.asElement(protoKey);
+};
+
+helpers.ensureTemplateSize = async function (b64Template, screenWidth, screenHeight) {
+  let {width: tplWidth, height: tplHeight, imgObj} = await getJimpData(b64Template);
+
+  // if the template fits inside the screen dimensions, we're good
+  if (tplWidth <= screenWidth && tplHeight <= screenHeight) {
+    return b64Template;
+  }
+
+  // otherwise, scale it to fit inside the screen dimensions
+  imgObj = imgObj.scaleToFit(screenWidth, screenHeight);
+  return (await imgObj.getBufferPromise(Jimp.MIME_PNG)).toString('base64');
+};
+
+helpers.getScreenshotForImageFind = async function (screenWidth, screenHeight) {
+  let b64Screenshot = await this.getScreenshot();
+  let {width: shotWidth, height: shotHeight, imgObj} = await getJimpData(b64Screenshot);
+
+  if (screenWidth === shotWidth && screenHeight === shotHeight) {
+    // the height and width of the screenshot and the device screen match, which
+    // means we should be safe when doing template matches
+    return b64Screenshot;
+  }
+
+  // otherwise, if they don't match, it could spell problems for the accuracy
```

Comment:
I would rather make this configurable from settings to make it as performant as possible by default and only do these transformations if explicitly asked by the client